### PR TITLE
Sections API: Don't let admins list _all_ sections

### PR DESCRIPTION
**Pull request by islemaster**
The sections list isn't showing up for admin users on production.  After writing a few unit tests I realized that this is because the API requests from admins are trying to load _every single section_ :scream: on prod and return its summary via the API, rather than only the admin user's own sections.  This change makes the API only return the admin user's sections when they `GET /dashboardapi/sections`.

I didn't realize the `load_and_authorize_resource` helper would have this side effect for admin users, who we've granted all permissions (with a few exceptions) using `CanCan`.  This is a potentially dangerous side effect that might exist elsewhere in our system - we should be careful when browsing the site as admin users, and make sure we write admin-specific unit tests for our controllers.

The bug only affects admins (internal) though so probably doesn't merit a hotfix.

### CODE REVIEW

File: dashboard/app/models/ability.rb [hunk 1]
```diff
@@ -184,6 +184,11 @@ def initialize(user)
     if user.admin?
       can :manage, :all
 
+      # We don't want admins to accidentally try to index _all_ existing
+      # sections so constrain their access.
+      cannot :index, Section
+      can :index, Section, user_id: user.id
```
Comment (islemaster) [inline dashboard/app/models/ability.rb]: It'd be a bigger change, but it might be safest to remove `:index` permission for everything from admins and explicitly restore it in places we want it, with appropriate constraints.

```ruby
if user.admin?
  can :manage, :all

  # Explicitly enumerate indexable resources to avoid admins accidentally causing slow queries
  cannot :index, :all
  can :index, Section, user_id: user.id
  can :index, Script
  #...etc
```
If we would like to pursue this approach it should be a separate PR with some manual testing around the result.

Comment (islemaster) [reply dashboard/app/models/ability.rb]: Also, read [Ability Precedence](https://github.com/ryanb/cancan/wiki/Ability-Precedence) in the CanCan docs to understand why the 'contradictions' in this specification work.

File: dashboard/app/controllers/api/v1/sections_controller.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class Api::V1::SectionsController < Api::V1::JsonApiController
   # GET /api/v1/sections
   # Get the set of sections owned by the current user
   def index
-    render json: @sections.map(&:summarize)
+    render json: current_user.sections.map(&:summarize)
```
Comment (islemaster) [inline dashboard/app/controllers/api/v1/sections_controller.rb:9]: New approach: Instead of messing with admin permissions in ability.rb (although I still think there might be merit to that generally) we fix the implementation of our controller to accurately return "sections owned by the current user," rather than "sections the current user has permission to index" which is what leaning on CanCan(Can) `load_and_authorize_resource` was doing.

Comment (aoby) [reply dashboard/app/controllers/api/v1/sections_controller.rb:9]: I agree it'd be worth exploring limiting the `index` permission for admins in CanCan as a follow up (or tech debt item).
